tempesta_fw/http_frame.h

```diff
@@ -0,0 +1,116 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2014 NatSys Lab. (info@natsys-lab.com).
+ * Copyright (C) 2015-2019 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+#ifndef __HTTP_FRAME__
+#define __HTTP_FRAME__
+
+#include "connection.h"
+
+#define FRAME_HEADER_SIZE	9
+
+/**
+ * FSM states for HTTP/2 frames processing.
+ */
+typedef enum {
+	HTTP2_RECV_FRAME_HEADER,
+	HTTP2_RECV_CLI_START_SEQ,
+	HTTP2_RECV_FIRST_SETTINGS,
+	HTTP2_RECV_FRAME_SERVICE,
+	HTTP2_RECV_FRAME_SETTINGS,
+	HTTP2_RECV_FRAME_GOAWAY,
+	HTTP2_RECV_FRAME_PADDED,
+	HTTP2_RECV_HEADER_PRI,
+	HTTP2_IGNORE_FRAME_DATA,
+	__HTTP2_RECV_FRAME_APP,
+	HTTP2_RECV_HEADER = __HTTP2_RECV_FRAME_APP,
+	HTTP2_RECV_CONT,
+	HTTP2_RECV_DATA
+} TfwFrameState;
+
+/**
+ * Unpacked header data of currently processed frame (RFC 7540 section
+ * 4.1). Reserved bit is not present here since it has no any semantic
+ * value for now and should be always ignored.
+ *
+ * @length	- the frame's payload length;
+ * @stream_id	- id of current stream (which frame is processed);
+ * @type	- the type of frame being processed;
+ * @flags	- frame's flags;
+ */
+typedef struct {
+	int		length;
```

Comment:
Never going to be below 0, thus should be unsigned.

Reply:
> Never going to be below 0

Currently, code in `tfw_http2_recv_priority()` may temporarily make `length` negative. If, for example, client sends a wrong length in PRIORITY frame.